docs/cpu_templates/cpu-template-helper.md

```diff
@@ -0,0 +1,205 @@
+the guest CPU configuration can vary depending on the infrastructure, updating
+it could lead to a situation where a custom CPU template loses its validity.
+In addition, even if values of the guest CPU configuration don't change, its
+internal behavior or semantics could still change. For instance, a kernel
+version update may introduce changes to KVM emulation and a microcode update may
+alter the bahavior of CPU instructions.
+
+To ensure awareness of these changes, it is strongly recommended to store the
+fingerprint file at the time of creating a custom CPU template and to
+continuously compare it with the current one.
+
+#### Compare command
+This command compares two fingerprint files: one was taken at the time of custom
+CPU template creation and the other is taken currently.
+```
+cpu-template-helper fingerprint compare --prev <prev-fingerprint> --curr <curr-fingerprint> --filters <field-1> [..<field-N>]
+```
+
+By continously comparing fingerprint files, users can ensure they are aware of
+any changes that could require revising the custom CPU template. However, it is
+worth noting that not all of these changes necessarily require a revision, and
+some detected changes could be false alarms. To provide users with flexibility
+in comparing fingerprint files based on situations or use cases, the `--filters`
+option allows users to select which fields to compare.
+
+As examples of when to compare fingerprint files:
+* When bumping the firecracker version up
+* When bumping the kernel version up
+* When applying a microcode update (or launching a new AWS EC2 instance)
+
+## Sample scenario
+This section gives steps of creating and managing a custom CPU template in a
+sample scenario where the template is designed to provide a consistent set of
+CPU features to a heterogeneous fleet consisting of multiple CPU models.
+
+### Custom CPU template creation
+1. Run the `cpu-template-helper template dump` command on each CPU model to
+retrieve guest CPU configuration.
+1. Run the `cpu-template-helper template strip` command to remove identical
+entries across the dumped guest CPU configuration files.
+1. Examine the differences of guest CPU configuration in details, determine
+which CPU features should be presented to guests and draft a custom CPU
+template.
+1. Run the `cpu-template-helper template verify` command to check the created
+custom CPU template is applied correctly.
+1. Conduct thorough testing of the template as needed to ensure that it does
+not contain any inconsistent entries and does not leaf to guest crashes.
+
+### Custom CPU template management
+1. Run the `cpu-template-helper fingerprint dump` command on each CPU model
+at the same time when creating a custom CPU template.
+1. Store the dumped fingerprint files together with the custom CPU template.
+1. Run the `cpu-template-helper fingerprint dump` command to ensure the
+template's validity whenever necessary.
```

Comment:
`whenever necessary` - we can recommend doing it when users expect that changes to the environment (hw/sw stack) may occur. Maybe suggest them review their update process to help identify such points?